### ISSUE TITLE
Multiple tenants experience error while executing Storage Account Analysis with v6.3.4

### ISSUE THREAD

**Issue by MarcelHeek** [JulianHayward/Azure-MG-Sub-Governance-Reporting#218]
**AzGovViz version**
v6.3.4

**CodeRunPlatform**
GitHub Actions

**Describe the bug**
Multiple tenants throw error processing the XMLSAProperties as can be seen in the screenshot below

**Screenshots**
![image](https://github.com/JulianHayward/Azure-MG-Sub-Governance-Reporting/assets/84451910/e341601f-af06-4861-88d2-c575a8e5a14d)

InvalidArgument: 
Line |
  50 |  …                 Write-Host $($saProperties.ForEach({ [char]$_ }) -joi …
     |                                                         ~~~~~~~~
     | Cannot convert value "﻿<?xml version="1.0" encoding="utf-8"?><StorageServiceProperties><Logging><Version>1.0</Version><Read>false</Read><Write>false</Write><Delete>false</Delete><RetentionPolicy><Enabled>false</Enabled></RetentionPolicy></Logging><HourMetrics><Version>1.0</Version><Enabled>true</Enabled><IncludeAPIs>true</IncludeAPIs><RetentionPolicy><Enabled>true</Enabled><Days>7</Days></RetentionPolicy></HourMetrics><MinuteMetrics><Version>1.0</Version><Enabled>false</Enabled><RetentionPolicy><Enabled>false</Enabled></RetentionPolicy></MinuteMetrics><Cors /><DeleteRetentionPolicy><Enabled>false</Enabled><AllowPermanentDelete>false</AllowPermanentDelete></DeleteRetentionPolicy></StorageServiceProperties>" to type "System.Char". Error: "String must be exactly one character long."
Error: Error: The process '/usr/bin/pwsh' failed with exit code 1

**Additional context**

**Comment by JulianHayward** [comment]
@MarcelHeek is this a new SA (did it work for this one before), any special configuration?

**Comment by MarcelHeek** [comment]
@JulianHayward 

I looked back 1 month ago (for one of the tenants) and the logic was using v6.3.0 and there was no issue at that time and I looked at the activity log of the SA and it was created >2 months ago.

I cannot see anything in the configuration that makes me frown and think of any blocking settings.

**Comment by MarcelHeek** [comment]
@JulianHayward I tested with v6.3.0 and the same issue. I must say I have not checked the execution recently due to circumstances, but the latest successful runs were Nov 14th and was while on v6.3.0. So this makes it even more strange

**Comment by kaiaschulz** [comment]
Hey,
@MarcelHeek here we go: [https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/kaiaschulz-6.3.5/pwsh/AzGovVizParallel.ps1](https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/kaiaschulz-6.3.5/pwsh/AzGovVizParallel.ps1)

**Comment by MarcelHeek** [comment]
@kaiaschulz The datacollection now succeeds for a subset of our tenants, there is one tenant with similar issue but now form the blobcontainer. Perhaps this needs the same fix?

![image](https://github.com/JulianHayward/Azure-MG-Sub-Governance-Reporting/assets/84451910/b8393ea8-fba1-4a2b-b720-0396bda0e205)

**Comment by kaiaschulz** [comment]
Hey @MarcelHeek ,
thanks for testing and providing the feedback.

Have you used [my provided code with v6.3.5](https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/kaiaschulz-6.3.5/pwsh/AzGovVizParallel.ps1) or did you just replaced the suggested code snipped?
Because I had exactly the same problem after my first try.
Afterwards I figured out that the failing script block was exchanged on 2 positions:
```POWERSHELL
# $xmlSaProperties = [xml]([string]$saProperties -replace $saProperties.Substring(0, 3)) # Leading character: ï»¿ (PS version <= 7.3.9)
# $xmlSaProperties = [xml]([string]$saProperties -replace $saProperties.Substring(0, 1)) # Leading character: ﻿ or U+feff (PS version >= 7.4.0)
$xmlSaProperties = [xml]($saProperties -replace ($saProperties -replace "<.*")) # Universal fix for all PS versions
```
and
```POWERSHELL
# $xmlListContainers = [xml]([string]$listContainers -replace $listContainers.Substring(0, 3)) # Leading character: ï»¿ (PS version <= 7.3.9)
# $xmlListContainers = [xml]([string]$listContainers -replace $listContainers.Substring(0, 1)) # Leading character: ﻿ or U+feff (PS version >= 7.4.0)
$xmlListContainers = [xml]($listContainers -replace ($listContainers -replace "<.*")) # Universal fix for all PS versions
```

I successfully ran [this script v6.3.5](https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/kaiaschulz-6.3.5/pwsh/AzGovVizParallel.ps1) within 2 tenants and also the `$xmlListContainers` was working as expected.

**Comment by MarcelHeek** [comment]
@kaiaschulz I used your ref in my workflow, not the snippets. I run this within Github action context with Managed Identities on self-hosted Github runners as security principals for data collection, so setting up something locally for troubleshooting / debugging is not trivial, but I will try to arrange stuff.

**Comment by MarcelHeek** [comment]
Ok, I have some results to share. For a SA that runs without error the variable $saProperties 

`$saProperties = AzAPICall -AzAPICallConfiguration $azAPICallConf -uri $urlServiceProps -method 'GET' -listenOn 'Content' -currentTask "$($storageAccount.SA.name) get restype=service&comp=properties" -saResourceGroupName $resourceGroupName -unhandledErrorAction Continue`

gets populated with

`∩╗┐<?xml version="1.0" encoding="utf-8"?><StorageServiceProperties>.............</StorageServiceProperties>`

However the SA that generates an error the $saProperties is populated with numbers

![image](https://github.com/JulianHayward/Azure-MG-Sub-Governance-Reporting/assets/84451910/fecc0de5-92eb-4ac8-96a2-8d912e67dfec)

So typecasting that to XML document format fails.

One of the most obvious differences between the two SA's is that the faulted one is a "general purpose V1" and the other a GPV2 sku.

**Comment by JulianHayward** [comment]
@kaiaschulz and @MarcelHeek first: thanks a ton for your investment!

I cannot reproduce the issue with the numbers return in any of my tenants (where I think I have all possible variants of SA).
According to [docs](https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-service-properties) we should not get these numbers.. 
I am seeing tenants with successful execution with PS7.4.0
let´s dig deeper on the impacted SAs.. May I ask you to test the below (we only require READ permissions) if possible one time with PS7.4.0 and another one with a lower version?

```Powershell
#storageAccount REST ARM
$storageAccountName = 'saName'
$storageAccountResourceGroupName = 'saRG'
$storageAccountSubscriptionId = 'subId'	
$storageAccountARMURI = "https://management.azure.com/subscriptions/$($storageAccountSubscriptionId)/resourcegroups/$($storageAccountResourceGroupName)/providers/Microsoft.Storage/storageAccounts/$($storageAccountName)?api-version=2021-09-01"
$storageAccount = Invoke-AzRestMethod -uri $storageAccountARMURI -Method GET
$storageAccountBlobEndpoint = ($storageAccount.Content | ConvertFrom-Json).properties.primaryEndpoints.blob
Write-Host "`$storageAccountBlobEndpoint: $($storageAccountBlobEndpoint)"

#storageAccount REST Storage
#ref https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-service-properties
$azContext = Get-AzContext
$tokenRequestEndPoint = 'https://storage.azure.com'
$bearerToken = ([Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate($azContext.Account, $azContext.Environment, $azContext.Tenant.id.ToString(), $null, [Microsoft.Azure.Commands.Common.Authentication.ShowDialog]::Never, $null, $tokenRequestEndPoint)).AccessToken
$Header = @{
    'Content-Type'  = 'application/json';
    'x-ms-version'  = '2021-04-10';
    'Authorization' = "Bearer $($bearerToken)"
}

$uri = "$($storageAccountBlobEndpoint)?restype=service&comp=properties"
$res = Invoke-WebRequest -uri $uri -Headers $Header -Method GET
$res
$res.Content
try {
    $xml = [xml]($res.Content -replace '^.*?<', '<')
    Write-Host 'XML' -ForegroundColor Green
    $xml.gettype().Name
}
catch {
    Write-Host 'failed..' -ForegroundColor DarkRed
    $res.Content.gettype()
}
```

**Comment by JulianHayward** [comment]
and if possible try the following for the 'numbers

```
$byteArray = [byte[]] ($res.Content | convertfrom-json)
[System.Text.Encoding]::UTF8.GetString($byteArray)
```

**Comment by MarcelHeek** [comment]
@JulianHayward 
I get exactly the same results as @kaiaschulz  (tested with PS v7.4.0)

**Comment by kaiaschulz** [comment]
Hey @MarcelHeek,
after a lot of investigation, I have updated my forked repo: [https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/master/pwsh/AzGovVizParallel.ps1](https://github.com/kaiaschulz/Azure-MG-Sub-Governance-Reporting/blob/master/pwsh/AzGovVizParallel.ps1)

Could you might check if it is running within your environment?
In 2 of my tenants it is working without issues.

Thanks in advance.

**Comment by MarcelHeek** [comment]
@kaiaschulz Nothing less than successful executions on the 4 tenants here. Good job 👏

**Comment by JulianHayward** [comment]
awesome!

**Comment by JulianHayward** [comment]
fixed with 6.3.5
